### PR TITLE
port indexing, reindex asvs from pandas

### PR DESCRIPTION
**Pull request by jbrockmendel**
## What do these changes do?

- [ ] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [ ] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #? 
- [ ] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date 
- [ ] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

File: asv_bench/benchmarks/pandas/indexing.py [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+
+
+class AssignTimeseriesIndex:
+    def setup(self):
+        N = 100000
+        idx = date_range("1/1/2000", periods=N, freq="H")
+        self.df = DataFrame(np.random.randn(N, 1), columns=["A"], index=idx)
+
+    def time_frame_assign_timeseries_index(self):
+        self.df["date"] = self.df.index
```
Comment (Garra1980) [inline asv_bench/benchmarks/pandas/indexing.py:263]: Is this somehow particularly important to track?

File: asv_bench/benchmarks/pandas/indexing.py [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+    UInt64Index,
+    concat,
+    date_range,
+    period_range,
+)
+
+import pandas._testing as tm
+
+
+class NumericSeriesIndexing:
```
Comment (Garra1980) [inline asv_bench/benchmarks/pandas/indexing.py:25]: As far as I understand we have some of these already, let's leave only really valuable ones

File: asv_bench/benchmarks/pandas/indexing.py [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+        indexes = {
+            "int": Int64Index(np.arange(N)),
+            "datetime": date_range("2011-01-01", freq="S", periods=N),
+        }
+        index = indexes[index]
+        self.s = Series(np.random.rand(N), index=index)
+        self.indexer = np.random.randint(0, N, size=N)
+
+    def time_take(self, index):
+        self.s.take(self.indexer)
```
Comment (Garra1980) [inline asv_bench/benchmarks/pandas/indexing.py:138]: Let's drop this if Modin implementation uses loc/iloc under the hood

File: asv_bench/benchmarks/pandas/indexing.py [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+        self.data.loc[[800000]]
+
+    def time_loc_scalar(self, index, index_structure):
+        self.data.loc[800000]
+
+    def time_loc_slice(self, index, index_structure):
+        self.data.loc[:800000]
+
+
+class DataFrameNumericIndexing:
```
Comment (Garra1980) [inline asv_bench/benchmarks/pandas/indexing.py:85]: Same as comment for NumericSeriesIndexing

File: asv_bench/benchmarks/pandas/indexing.py (hunk 1 above)
Comment (jbrockmendel) [reply asv_bench/benchmarks/pandas/indexing.py:263]: In essentially every case the answer is that "pandas implemented this for a reason"

File: asv_bench/benchmarks/pandas/indexing.py (hunk 2 above)
Comment (jbrockmendel) [reply asv_bench/benchmarks/pandas/indexing.py:25]: Please be more specific.

Comment (Garra1980) [reply asv_bench/benchmarks/pandas/indexing.py:25]: My concern is only to avoid benchmarking duplication

Comment (Garra1980) [reply asv_bench/benchmarks/pandas/indexing.py:25]: I let @YarShev and @anmyachev decide whether we can/should skip some cases

Comment (YarShev) [reply asv_bench/benchmarks/pandas/indexing.py:25]: I would suggest to integrate new tests directly into our asv benchmarking system so we can see the difference and new added cases.

**Pull request closed without merging**